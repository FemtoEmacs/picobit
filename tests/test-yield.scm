;; tests yield

(define (infinity)
  (let loop ((a 1))
    (begin 
      (yield a)
      (loop (+ a 1)))))

(display (infinity))
(display (infinity))
(display (infinity))
(display (infinity))
(display (infinity))
(display (infinity))
(display (infinity))
(display (infinity))
(display (infinity))
(display (infinity))
(display (infinity))
(display (infinity))
(display (infinity))
(display (infinity))