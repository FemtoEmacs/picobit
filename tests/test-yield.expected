:0080000080
:10800000FBD71D0481194000810F4000810C400006
:10801000810A4000810B4000804200028027011647
:1080200080270117802401188071011A806E011BBE
:108030008027011C8077011D8073011E80240002AF
:10804000806A00028073011F806701208040011058
:1080500080400111803201128027011380660114D3
:108060008069011580780002807D010D8066010E17
:10807000807001098078010902B4020651005002A3
:10808000B401E95202B401E353B001A8B5D0EEB097
:1080900001A8B5CAEEB001A8B5C4EEB001A8B5BE3E
:1080A000EEB001A8B5B8EEB001A8B5B2EEB001A827
:1080B000B5ACEEB001A8B5A6EEB001A8B5A0EEB083
:1080C00001A8B59AEEB001A8B594EEB001A8B58E9E
:1080D000EEB001A8B588EEB001A8B582EED80120B7
:1080E000DAB8954321DBB6800221CC9A21CE2161FA
:1080F000EE2022CFB67200EF20B6800120DA9EA0DB
:108100002607E620B558EEA02607E600EF20C0B807
:10811000BF20B582B648012004C8B8AEA03121B551
:10812000A402B583CDDCEF02A034210EC5C122CD5F
:108130002102CDB401377101210EC89220EF210E2A
:10814000416221B662010421C2EF2002B559B75540
:1081500020CCB8B8A02C07E620CEB51FEE20CFB6B5
:10816000800120D296A02D07E600EF20CC9DA02410
:1081700007E620CEB505EE20CFB666A106B000DE3C
:10818000EE20B000FBEEA02D07E600EF20D995A170
:1081900007B100DE20F09C2095A105B100DEA1040E
:1081A000B100DEA108B100DE000002B401AF710130
:1081B00020B5AB02B401B87101202102CDB401D2C7
:1081C000B58AEE0521B581710120CEEF022022D0C3
:1081D00000EF01204261EE2005C122B56B71012044
:1081E00002CDEF012007E600EF00D502B401F071E7
:1081F000014021D040CF5040CE02B401FE71014079
:1082000000D02000D7EF022004C8212302CDCDB436
:10821000021371012204C822242302CDCDCDB40261
:1082200022710123B8DC21B0014523222602CDCDE5
:10823000CDB40235710122B8C621B00145212426F2
:1082400002CDCDCDB4024871012022242602CDCD2D
:10825000CDCDB4025670002121C4232502CDCDB46A
:108260000263710122B893219220EF229801942099
:10827000B1014520EF21B87BB77500B86EB76A2110
:05828000B73B21B7250A
:00000001FF
